service-api/module/Application/test/Aws/Secrets/AwsSecretTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ApplicationTest\Aws\Secrets;
+
+use Application\Aws\Secrets\AwsSecret;
+use Application\Aws\Secrets\AwsSecretsCache;
+use PHPUnit\Framework\TestCase;
+use RuntimeException;
+
+class AwsSecretTest extends TestCase
+{
+    private AwsSecret $sut;
+
+    public function setUp(): void
+    {
+        $this->sut = new AwsSecret('test');
+    }
+
+    public function testGetName(): void
+    {
+        self::assertEquals('test', $this->sut->getName());
+    }
+
+    public function testGetValueThrowsExceptionIfCacheNotInstantiated(): void
+    {
+        $this->expectException(RuntimeException::class);
+        $this->expectExceptionMessage('SecretsCache has not been initialised');
+
+        $this->sut->getValue();
+    }
+
+    public function testGetValueReturnsValueFromCache(): void
+    {
+        $cache = $this->createMock(AwsSecretsCache::class);
+        $cache->expects(self::once())
+            ->method('getValue')
+            ->with('test')
+            ->willReturn('secretValue');
```

Comment:
Is `AwsSecretsCache->getValue` the right level to mock this at? At the moment you can't test if it goes to network or hits the cache. Just that the cache returns a value.